gordo_components/client/client.py

```diff
@@ -198,36 +227,51 @@ def _endpoints_from_watchman(self, endpoint: str) -> typing.List[EndpointMetadat
-                tag_list=normalize_sensor_tags(
-                    data["endpoint-metadata"]["metadata"]["dataset"]["tag_list"]
-                ),
-                target_tag_list=normalize_sensor_tags(
-                    data["endpoint-metadata"]["metadata"]["dataset"]["target_tag_list"]
-                ),
-                resolution=data["endpoint-metadata"]["metadata"]["dataset"][
-                    "resolution"
-                ],
-                model_offset=data["endpoint-metadata"]["metadata"]["model"].get(
-                    "model-offset", 0
-                ),
-            )
-            if data["healthy"]
-            else EndpointMetadata(
-                target_name=None,
-                healthy=data["healthy"],
-                endpoint=f'{self.base_url}{data["endpoint"].rstrip("/")}',
-                tag_list=None,
-                target_tag_list=None,
-                resolution=None,
-                model_offset=None,
-            )
-            for data in resp.json()["endpoints"]
-        ]
+        # Check if we're in a modified Client, and have previously cached the endpoints
+        if hasattr(self, "__endpoints") and getattr(self, "__endpoints") is not None:
+            endpoints = getattr(self, "__endpoints")
+
+        # Otherwise get the endpoints from watchman.
+        else:
+            endpoints = [
+                EndpointMetadata(
+                    target_name=data["endpoint-metadata"]["metadata"]["name"],
+                    healthy=data["healthy"],
+                    endpoint=f'{self.base_url}{data["endpoint"].rstrip("/")}',
+                    tag_list=normalize_sensor_tags(
+                        data["endpoint-metadata"]["metadata"]["dataset"]["tag_list"]
+                    ),
+                    target_tag_list=normalize_sensor_tags(
+                        data["endpoint-metadata"]["metadata"]["dataset"][
+                            "target_tag_list"
+                        ]
+                    ),
+                    resolution=data["endpoint-metadata"]["metadata"]["dataset"][
+                        "resolution"
+                    ],
+                    model_offset=data["endpoint-metadata"]["metadata"]["model"].get(
+                        "model-offset", 0
+                    ),
+                )
+                if data["healthy"]
+                else EndpointMetadata(
+                    target_name=None,
+                    healthy=data["healthy"],
+                    endpoint=f'{self.base_url}{data["endpoint"].rstrip("/")}',
+                    tag_list=None,
+                    target_tag_list=None,
+                    resolution=None,
+                    model_offset=None,
+                )
+                for data in resp.json()["endpoints"]
+            ]
+
+            # If we're in a modified Client version, we will cache these endpoints
+            # into the class itself.
+            if hasattr(self, "__endpoints"):
```

Comment:
Again, maybe (probably) just me not used to the reflection. Seems to be oh so popular.